pkg/apis/kubeone/types.go

```diff
@@ -395,8 +399,28 @@ type ClusterNetworkConfig struct {
 
 	// KubeProxy config
 	KubeProxy *KubeProxyConfig `json:"kubeProxy,omitempty"`
+
+	// IPFamily IPv4 | IPv6 | IPv4+IPv6 | IPv6+IPv4.
+	IPFamily IPFamily `json:"ipFamily,omitempty"`
+
+	// NodeCIDRMaskSizeIPv4 is the mask size used to address the nodes within provided IPv4 Pods CIDR. It has to be larger than the provided IPv4 Pods CIDR. Defaults to 24.
+	NodeCIDRMaskSizeIPv4 *int `json:"nodeCIDRMaskSizeIPv4,omitempty"`
+
+	// NodeCIDRMaskSizeIPv6 is the mask size used to address the nodes within provided IPv6 Pods CIDR. It has to be larger than the provided IPv6 Pods CIDR. Defaults to 64.
+	NodeCIDRMaskSizeIPv6 *int `json:"nodeCIDRMaskSizeIPv6,omitempty"`
 }
 
+// IPFamily allows specifying IP family of a cluster.
+// Valid values are IPv4 | IPv6 | IPv4+IPv6 | IPv6+IPv4.
+type IPFamily string
+
+const (
+	IPFamilyIPv4     IPFamily = "IPv4"      // IPv4 only cluster.
+	IPFamilyIPv6     IPFamily = "IPv6"      // IPv6 only cluster.
+	IPFamilyIPv4IPv6 IPFamily = "IPv4+IPv6" // Dualstack cluster with IPv4 as primary address family.
+	IPFamilyIPv6IPv4 IPFamily = "IPv6+IPv4" // Dualstack cluster with IPv6 as primary address family.
```

Comment:
Can you put those comments above variables so that we have valid godoc-comments?